// telegram-types/generate-dts.ts
import { writeFileSync } from "fs";
import types from "./types";
const interfaces: IInterface[] = types.types.map((type) => {
    return {
        name: type.name,
        description: type.description,
        fields: type.fields.map((fieldType) => {
            return {
                name: fieldType.name,
                description: fieldType.type === "Integer" ? "Integer. " : "" + fieldType.description,
                required: !fieldType.optional,
                type: tmTypeToTs(fieldType.type),
            };
        }),
    };
});
const content = interfaces.map((i) => interfaceToString(i)).join("\n");
writeFileSync(__dirname + "/index.d.ts", content);
// tslint:disable-next-line:no-console
console.log("Write successfully " + interfaces.length);

export function interfaceToString(iface: IInterface) {
    return `// ${iface.description}
export interface ${iface.name} {
${iface.fields.map((field) => {
            return "\t" + field.name + (field.required ? "" : "?") + ": " + field.type + "; // " + field.description;
        }).join("\n")}
}
`;

}

export function tmTypeToTs(type: string) {
    switch (type) {
        case "String":
            return "string";
        case "Integer":
            return "number";
        case "Boolean":
            return "boolean";
        default:
            throw new Error("Unknown Telegram type: " + type);
    }
}

export interface IInterfaceField {
    name: string;
    description: string;
    type: string;
    required: boolean;
}
export interface IInterface {
    name: string;
    description: string;
    fields: IInterfaceField[];
}
