// telegram-types/types.ts
// tslint:disable:max-line-length
export default {
    types: [{
        name: "User",
        description: "This object represents a Telegram user or bot",
        fields: [{
            name: "id",
            type: "Integer",
            optional: false,
            description: "Unique identifier for this user or bot",
        }, {
            name: "first_name",
            type: "String",
            optional: false,
            description: "User‘s or bot’s first name",
        }, {
            name: "last_name",
            type: "String",
            optional: true,
            description: "User‘s or bot’s last name",
        }, {
            name: "username",
            type: "String",
            optional: true,
            description: "User‘s or bot’s username",
        }],
    }, {
        name: "Chat",
        description: "This object represents a chat",
        fields: [{
            name: "id",
            type: "Integer",
            optional: false,
            description: "Unique identifier for this chat. This number may be greater than 32 bits and some programming languages may have difficulty/silent defects in interpreting it. But it is smaller than 52 bits, so a signed 64 bit integer or double-precision float type are safe for storing this identifier",
        }, {
            name: "type",
            type: "String",
            optional: false,
            description: "Type of chat, can be either “private”, “group”, “supergroup” or “channel”",
        }, {
            name: "title",
            type: "String",
            optional: true,
            description: "Title, for supergroups, channels and group chats",
        }, {
            name: "username",
            type: "String",
            optional: true,
            description: "Username, for private chats, supergroups and channels if available",
        }, {
            name: "first_name",
            type: "String",
            optional: true,
            description: "First name of the other party in a private chat",
        }, {
            name: "last_name",
            type: "String",
            optional: true,
            description: "Last name of the other party in a private chat",
        }, {
            name: "all_members_are_administrators",
            type: "Boolean",
            optional: true,
            description: "True if a group has ‘All Members Are Admins’ enabled",
        }],
    }],
};
